File: Javascript/traversy-js-challenges-main/08-binary-trees-graphs/12-graph-implementation/graph.js
class Graph {
    constructor() {
        this.adjacencyList = {};
    }

    addVertex(vertex) {
        this.adjacencyList[vertex] = [];
    }

    addEdge(vertex1, vertex2) {
        this.adjacencyList[vertex2].push(vertex1);
        this.adjacencyList[vertex1].push(vertex2);
    }

    removeEdge(vertex1, vertex2) {
        this.adjacencyList[vertex2] = this.adjacencyList[vertex2].filter(
            (v) => v !== vertex1);
        this.adjacencyList[vertex1] = this.adjacencyList[vertex1].filter(
            (v) => v !== vertex2);
    }

    removeVertex(vertex) {
        while (this.adjacencyList[vertex].length) {
            const adjacentVertex = this.adjacencyList[vertex].pop();
            this.removeEdge(vertex, adjacentVertex);
          }
        delete this.adjacencyList[vertex];
    }


}

// Example Adjacency List
// {
//   'A': ['B', 'C'],
//   'B': ['A', 'D'],
//   'C': ['A', 'D'],
//   'D': ['B', 'C']
// };

module.exports = Graph;

File: Python/GraphPractice/adjacency_matrix_graph/graph.py

class Graph:

    def __init__(self):
        self.adjacencyListDict = {}

    def add_vertex(self, vertex):
        self.adjacencyListDict[vertex] = []

    def add_edge(self, vertex1, vertex2):
        self.adjacencyListDict[vertex1].append(vertex2)
        self.adjacencyListDict[vertex2].append(vertex1)

    def remove_edge(self, vertex1, vertex2):
        self.adjacencyListDict[vertex1] = filter(
            lambda item: item != vertex2, self.adjacencyListDict[vertex1]
        )
        self.adjacencyListDict[vertex2] = filter(
            lambda item: item != vertex1, self.adjacencyListDict[vertex2]
        )

    def remove_vertex(self, vertex):
        while len(self.adjacencyListDict[vertex]) > 0:
            adjacent_vertex = self.adjacencyListDict[vertex].pop()
            self.adjacencyListDict[adjacent_vertex] = filter(
                lambda item: item != vertex, self.adjacencyListDict[adjacent_vertex]
            )
        del self.adjacencyListDict[vertex]

